exporter/loadbalancingexporter/resolver_k8s_handler.go

```diff
@@ -44,15 +44,24 @@ func (h handler) OnAdd(obj interface{}, _ bool) {
 }
 
 func (h handler) OnUpdate(oldObj, newObj interface{}) {
-	switch oldObj.(type) {
+	switch oldEps := oldObj.(type) {
 	case *corev1.Endpoints:
+		epRemove := convertToEndpoints(oldEps)
+		for _, ep := range epRemove {
+			h.endpoints.Delete(ep)
+		}
+		if len(epRemove) > 0 {
+			_, _ = h.callback(context.Background())
+		}
+
 		newEps, ok := newObj.(*corev1.Endpoints)
```

Comment:
Should this check be at the beginning of the function, before the switch statement? Otherwise, it's possible that you'll delete all endpoints and end up returning before setting new endpoints.